bondjimbond/islandora_badges
# 7.x merged

Resolves #17 

# What does this Pull Request do?

Splits the Altmetrics, Scopus and Web of Science functionality into separate modules.

# What's new?

These modules are fairly simple and share very little except the DOI code. This makes it easier to add additional badges with a new module also depending on islandora_badges.

The islandora_badges module just holds the DOI Xpath configuration and a utility function to extract the DOI from an objects MODS record.

It was suggested by @DiegoPino that you could also allow for retrieving the DOI from Solr. So if someone has indexed it, you wouldn't have to parse the MODS each time.

# How should this be tested?

Make sure it still works as expected with objects.

Example:
* Does this change require documentation to be updated? Yes
* Does this change add any new dependencies? No
* Does this change require any other modifications to be made to the repository (ie. Regeneration activity, etc.)? No
* Could this change impact execution of existing code? No

# Interested parties
@mjordan @bondjimbond
